src/sunspec_model.cpp

```diff
@@ -208,8 +208,9 @@ int16_t Point::get_scale_factor() {
     EVLOG(debug) << "Attempting to read scale factor for point " << this->name << " with address offset " << this->offset;
 
     if (!this->requires_scale_factor_)
-        throw exceptions::scaling_error("Trying to read scale factor for a point that doesn't require one.");
-    
+        EVLOG(warning) << "Trying to read scale factor for a point that doesn't require one. Returning a neutral scale factor.\n";
+        return 0;
```

Comment:
I think that will always return 0 regardless of the condition. I would strongly suggest curly brackets around every if-else